cirq-core/cirq/sim/act_on_state_vector_args.py

```diff
@@ -174,9 +174,9 @@ def _perform_measurement(self, qubits: Sequence['cirq.Qid']) -> List[int]:
         )
         return bits
 
-    def _on_copy(self, target: 'cirq.ActOnStateVectorArgs', with_buffer: bool = True):
+    def _on_copy(self, target: 'cirq.ActOnStateVectorArgs', reuse_buffer: bool = False):
         target.target_tensor = self.target_tensor.copy()
-        if with_buffer:
+        if reuse_buffer:
```

Comment:
`if not reuse_buffer` here and above

Reply:
My understanding is: if the buffer is copied to the new object, then it can be "reused"; otherwise, the object might need to create its own buffer (from scratch). The point is, the context of the parameter is only the `copy` method. Its semantics should not be attached to a specific usage such as `_run`. What do you think?

Reply:
In act_on_args.copy in line 118, it does a shallow copy, meaning that if we don't do anything, then the buffer instance itself is not copied. Thus if `reuse_buffer` is set here, that means we don't have to do anything else. If `reuse_buffer` is false, then that means the calling function wants to forcibly deep-copy the buffer. So this condition needs changed to `if not reuse_buffer`.

I'm not quite sure I catch the question about semantics, but whether-or-not-to-copy-the-buffer is context-dependent, not data-structure-dependent. If the calling function is running repetitions serially (as `_run` does), then the buffer can be safely reused from one repetition to another. If the user is running them in parallel, then all referencing the same buffer is obviously unsafe. So it depends on the use case.